crates/core/tedge_mapper/src/c8y_converter.rs

```diff
@@ -18,12 +19,18 @@ const INVENTORY_FRAGMENTS_FILE_LOCATION: &str = "/etc/tedge/device/inventory.jso
 const INVENTORY_MANAGED_OBJECTS_TOPIC: &str = "c8y/inventory/managedObjects/update";
 const SUPPORTED_OPERATIONS_DIRECTORY: &str = "/etc/tedge/operations";
 const SMARTREST_PUBLISH_TOPIC: &str = "c8y/s/us";
+const TEDGE_ALARMS_TOPIC: &str = "tedge/alarms/";
+const INTERNAL_ALARMS_TOPIC: &str = "c8y-internal/alarms/";
 
 pub struct CumulocityConverter {
     pub(crate) size_threshold: SizeThreshold,
     children: HashSet<String>,
     pub(crate) mapper_config: MapperConfig,
     device_name: String,
+
+    syncing: bool,
+    pending_alarms_map: HashMap<String, Message>,
+    old_alarms_map: HashMap<String, Message>,
```

Comment:
This 3 fields are always used in concert with a systematic pattern `if syncing {do this with the maps} else {do that, the maps being used}`.

=> I would extract these fields in a separate struct and use an enum to differentiate the cases.

```
pub struct CumulocityConverter {
     ... 
     synchronizer: AlarmSynchronizer,
}

enum AlarmSynchronizer {
     Syncing { pending_alarms_map: HashMap<String, Message>, old_alarms_map: HashMap<String, Message> }
     SyncDone
}

impl AlarmSynchronizer  {
     fn try_convert_alarm(&mut self, input: &Message) -> Result<Vec<Message>, ConversionError> { }
     fn process_internal_alarm(&mut self, input: &Message) {}
     fn sync_messages(&mut self ) {}
}
```